### PR TITLE
Mqtt-kafka single group support cont

### LINKED ISSUE #698: `mqtt-kafka` binding uses 2 different consumer groups per `mqtt` client
**Describe the bug**
`mqtt-kafka` binding uses 2 different consumer groups per `mqtt` client (instead of just one).

**To Reproduce**
1. Run `mqtt.kafka.broker` example
2. Connect `mqtt` client
3. List consumer groups in Kafka
4. See multiple consumer groups created for the same `mqtt` client

**Expected behavior**
There should be only one consumer group for each `mqtt` client.

**Screenshots**
None

**Desktop (please complete the following information):**
N/A

**Smartphone (please complete the following information):**
N/A

**Additional context**
None

### PR DESCRIPTION
**Pull request by akrambek**
## Description

Mqtt-kafka single group support cont

Fixes #698

### CODE REVIEW

File: specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/application/produce/message.producer.id/server.rpt [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                              .build()}
+write flush
+
+accepted
+
+read zilla:begin.ext ${kafka:beginEx()
+                              .typeId(zilla:id("kafka"))
+                              .produce()
+                                  .topic("test")
+                                  .partition(0)
```
Comment (jfallows) [inline specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/application/produce/message.producer.id/server.rpt:56]: `producerId` and `producerEpoch`?

**Review by jfallows — COMMENTED** [review]
We also need scenario(s) to check the behavior when producer id changes on the same client produce stream, either to a different producer id value or to no producer id.

This will happen frequently if multiple clients are producing at a mixture of qos to the same kafka topic such as mqtt-messages.

File: specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/application/merged/unmerged.produce.message.values.producer.id/server.rpt [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+                                  .build()
+                              .build()}
+
+connected
+
+write zilla:begin.ext ${kafka:beginEx()
+                               .typeId(zilla:id("kafka"))
+                               .produce()
+                                   .topic("test")
+                                   .partition(1)
```
[the remark this answers is not in the document]
Comment (akrambek) [reply specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/application/merged/unmerged.produce.message.values.producer.id/server.rpt:169]: it is part of dataEx I have filled missing ones.  We can't send producerId and producerEpoch as part of beginEx since we don't know the partitions and we need partitions to fetch.

File: specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/application/merged/merged.produce.message.value.partition.id/client.rpt [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+                                   .partition(0, -1)
+                                   .build()
+                              .build()}
+
+write advise zilla:flush ${kafka:flushEx()
+                              .typeId(zilla:id("kafka"))
+                              .merged()
+                                .produce()
+                                  .key("a")
+                                  .hashKey("key7")
```
Comment (jfallows) [inline specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/application/merged/merged.produce.message.value.partition.id/client.rpt:46]: Does `produce` `flush` need both `key` and `hashKey` or just `hashKey`?

**Review by jfallows — CHANGES_REQUESTED** [review]
Looks pretty good.

Please add a negative test to reject gap in sequence for non-zero producerId in client produce factory.

Please add a test for replaying sequence from earlier value, but still no gaps.

File: specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/application/merged/merged.produce.message.value.partition.id/client.rpt (hunk 3 above)
Comment (akrambek) [reply specs/binding-kafka.spec/src/main/scripts/io/aklivity/zilla/specs/binding/kafka/streams/application/merged/merged.produce.message.value.partition.id/client.rpt:45]: yes

**Review by jfallows — APPROVED** [review]
